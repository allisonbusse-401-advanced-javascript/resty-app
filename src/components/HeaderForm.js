import React from 'react';
import PropTypes from 'prop-types';

const HeaderForm = ({ handleHeaderSubmit, handleChange, authUsername, authPassword, authToken }) => (
  <form onSubmit={handleHeaderSubmit}>
    <button>Headers</button>
    <div>
      <h2>Basic Authorization</h2>
      <input type="text" name="authUsername" placeholder="Username" value={authUsername} onChange={handleChange}></input>
      <input type="password" name="authPassword" placeholder="Password" value={authPassword} onChange={handleChange}></input>
    </div>
    <div>
      <h2>Bearer Token</h2>
      <input type="text" name="authToken" placeholder="Bearer Token" value={authToken} onChange={handleChange}></input>
    </div>
  </form>
);


HeaderForm.propTypes = {
  handleHeaderSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  authUsername:  PropTypes.string,
  authPassword: PropTypes.string,
  authToken: PropTypes.string,
};

export default HeaderForm;
