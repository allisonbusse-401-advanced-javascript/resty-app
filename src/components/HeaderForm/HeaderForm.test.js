import React from 'react';
import { shallow } from 'enzyme';
import HeaderForm from './HeaderForm';

describe('HeaderForm component', () => {
  const props = {
    handleHeaderSubmit: () => {},
    handleChange: () => {},
    authUsername: 'jkl',
    authPassword: 'jkl',
    authToken: 'uio'
  };
  it('renders HeaderForm', () => {   
    const wrapper = shallow(<HeaderForm {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
