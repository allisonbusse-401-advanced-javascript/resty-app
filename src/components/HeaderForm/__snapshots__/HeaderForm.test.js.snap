// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderForm component renders HeaderForm 1`] = `
<form
  onSubmit={[Function]}
>
  <button>
    Headers
  </button>
  <div>
    <h2>
      Basic Authorization
    </h2>
    <input
      name="authUsername"
      onChange={[Function]}
      placeholder="Username"
      type="text"
      value="jkl"
    />
    <input
      name="authPassword"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="jkl"
    />
  </div>
  <div>
    <h2>
      Bearer Token
    </h2>
    <input
      name="authToken"
      onChange={[Function]}
      placeholder="Bearer Token"
      type="text"
      value="uio"
    />
  </div>
</form>
`;
